version_template.py

```diff
@@ -107,14 +102,34 @@ def buildNumber():
 
 
 def currentCommit():
+    import os
     from subprocess import check_output
-    return check_output('git log --pretty=oneline -n 1 -- $(pwd)', shell=True).split()[0]
+
+    return check_output(['git', 'log', '--pretty=oneline', '-n', '1', os.getcwd()]).split()[0]
 
 
 def dirty():
+    """
+    Returns True if the code has uncommitted diffs.
+
+    :return: bool
+    """
     from subprocess import call
-    return 0 != call('(git diff --exit-code '
-                     '&& git diff --cached --exit-code) > /dev/null', shell=True)
+    import os
+
+    # pipe stdout to devnull since git diff can generate unnecessary output, and we only want return codes
+    with open(os.devnull, 'w')  as FNULL:
+        # return code is non-zero if there are still changes relative to the index
+        # otherwise return code is 0
+        retcode1 = call(['git', 'diff', '--exit-code'], stdout=FNULL)
```

Comment:
Consider [inlining](https://refactoring.com/catalog/inlineTemp.html) retcode1 and recode2. It will reduce the # of lines *and* the line length in this case.

Reply:
Done!  I agree, it looks much better.